src/model/entities/rundown.ts

```diff
@@ -590,9 +590,14 @@ export class Rundown extends BasicRundown {
 
     const oldSegment: Segment = this.segments[segmentIndex]
     if (oldSegment.isOnAir()) {
-      segment.setParts(oldSegment.getParts())
+      const newOnAirPart: Part | undefined = segment.getParts().find(part => part.isOnAir())
+      if (newOnAirPart) {
+        this.activeCursor = this.createCursor(this.activeCursor, { part: newOnAirPart })
+      }
+
       segment.putOnAir()
       this.activeCursor = this.createCursor(this.activeCursor, { segment })
+      this.updateInfinitePieces()
```

Comment:
I assume that it has already been thought through, but won't this create logic branches where the on air content can get updated by ingest changes?

Reply:
Potentially yes, but that is what Jacob wants for KLAR_ON_AIR